substratevm/src/com.oracle.svm.core.posix/src/com/oracle/svm/core/posix/JavaUtilZipSubstitutions.java

```diff
@@ -209,62 +228,111 @@ private int deflateBytes(long addr, byte[] b, int off, int len, int flush) {
         }
     }
 
+    // (JDK 11)src/java.base/share/native/libzip/Deflater.c-static jlong doDeflate(JNIEnv *env, jobject this, jlong addr,
+    //                       jbyte *input, jint inputLen,
+    //                       jbyte *output, jint outputLen,
+    //                       jint flush, jint params)
+    private static long doDeflate(long addr, CCharPointer input, int inputLen, CCharPointer output, int outputLen, int flush, int params) {
+        z_stream strm = WordFactory.pointer(addr);
+        int inputUsed, outputUsed;
+        int finished = 0;
+        int setParams = params & 1;
+
+        strm.set_next_in(input);
+        strm.set_next_out(output);
+        strm.set_avail_in(inputLen);
+        strm.set_avail_out(outputLen);
+
+        if (setParams != 0) {
+            int strategy = (params >> 1) & 3;
+            int level = params >> 3;
+            int res = ZLib.deflateParams(strm, level, strategy);
+            if (res == ZLib.Z_OK() || res == ZLib.Z_BUF_ERROR()) {
+                if (res == ZLib.Z_OK()) {
+                    setParams = 0;
+                }
+            } else {
+                throw new InternalError();
+            }
+        } else {
+            int res = ZLib.deflate(strm, flush);
+            if (res == ZLib.Z_OK() || res == ZLib.Z_STREAM_END() || res == ZLib.Z_BUF_ERROR()) {
+                if (res == ZLib.Z_STREAM_END()) {
+                    finished = 1;
+                }
+            } else {
+                throw new InternalError();
+            }
+        }
+        inputUsed = inputLen - strm.avail_in();
+        outputUsed = outputLen - strm.avail_out();
+        return ((long) inputUsed) | (((long)outputUsed) << 31) | (((long)finished) << 62) | (((long)setParams) << 63);
```

Comment:
All the style guidelines I can find say there should be a space after the cast.  And having a space after the first is just weird.